app/src/ui/branches/branch-select.tsx

```diff
@@ -0,0 +1,191 @@
+      return
+    }
+
+    const windowHeight = window.innerHeight
+    const bottomOfButton = this.invokeButtonRef.getBoundingClientRect().bottom
+    const listHeaderHeight = 75
+    const calcMaxHeight = Math.round(
+      windowHeight - bottomOfButton - listHeaderHeight
+    )
+
+    const dropdownListHeight =
+      calcMaxHeight > maxDropdownListHeight
+        ? maxDropdownListHeight
+        : calcMaxHeight
+    if (dropdownListHeight !== this.state.dropdownListHeight) {
+      this.setState({ dropdownListHeight })
+    }
+  }
+
+  private onInvokeButtonRef = (buttonRef: HTMLButtonElement | null) => {
+    this.invokeButtonRef = buttonRef
+  }
+
+  private toggleBranchDropdown = () => {
+    this.setState({ showBranchDropdown: !this.state.showBranchDropdown })
+  }
+
+  private closeBranchDropdown = () => {
+    this.setState({ showBranchDropdown: false })
+  }
+
+  private renderBranch = (item: IBranchListItem, matches: IMatches) => {
+    return renderDefaultBranch(item, matches, this.props.currentBranch)
+  }
+
+  private onItemClick = (branch: Branch, source: ClickSource) => {
+    source.event.preventDefault()
+    this.setState({ showBranchDropdown: false, selectedBranch: branch })
+    this.props.onChange?.(branch)
+  }
+
+  private onFilterTextChanged = (filterText: string) => {
+    this.setState({ filterText })
+  }
+
+  public renderBranchDropdown() {
+    if (!this.state.showBranchDropdown) {
+      return
+    }
+
+    const { currentBranch, defaultBranch, recentBranches, allBranches } =
+      this.props
+
+    const { filterText, selectedBranch, dropdownListHeight } = this.state
+
+    return (
+      <Popover
+        className="branch-select-dropdown"
+        caretPosition={PopoverCaretPosition.TopLeft}
+        onClickOutside={this.closeBranchDropdown}
+      >
+        <div className="branch-select-dropdown-header">
+          Choose a base branch
+          <button
+            className="close"
+            onClick={this.closeBranchDropdown}
+            aria-label="close"
+          >
+            <Octicon symbol={OcticonSymbol.x} />
+          </button>
+        </div>
+        <div
+          className="branch-select-dropdown-list"
+          style={{ height: `${dropdownListHeight}px` }}
+        >
+          <BranchList
+            allBranches={allBranches}
+            currentBranch={currentBranch}
+            defaultBranch={defaultBranch}
+            recentBranches={recentBranches}
+            filterText={filterText}
+            onFilterTextChanged={this.onFilterTextChanged}
+            selectedBranch={selectedBranch}
+            canCreateNewBranch={false}
+            renderBranch={this.renderBranch}
+            onItemClick={this.onItemClick}
+          />
+        </div>
+      </Popover>
+    )
+  }
+
+  public render() {
+    return (
+      <div className="branch-select-component">
+        <Button
+          onClick={this.toggleBranchDropdown}
+          onButtonRef={this.onInvokeButtonRef}
```

Comment:
If you just want to store it, I think you can do just this:
```suggestion
          onButtonRef={this.invokeButtonRef}
```

Reply:
I just verified that his does not work here because it is not a true `ref` as the `onButtonRef` is a callback for the `ref` on the `button` (lowercase) down in the wrapper `Button` component (uppercase);